### DISCUSSION TITLE
Decouple model distribution from kepler releases

### DISCUSSION THREAD

**Discussion by sthaha** [sustainable-computing-io/kepler#1245] [Ideas]
## Problem Statement

Currently Kepler container ships models that are used by the the `estimator`. Any updates to the model requires a new release of kepler. This tight coupling hinders new release of model or does not permit updating existing models of an already deployed `kepler`

## Proposal

Remove tight coupling by injecting model into kepler container using configmaps (on k8s) or volumemounts on pure containers. 

### Changes to Kepler builds
- [ ]  remove models from being added in Dockerfile
- [ ]  modify `build-manifests` target to add models as configmap mounted to `kepler` daemonset

### Model Releases
- [ ] Model releases can make use of Github artifacts to publish all models as an archive

### Kepler Operator changes
- [ ] Modify kepler-operator to include a configmap that ships models 
- [ ] Allow users to configure the configmap that contains models to be specified as part of (KeplerInternal at first, then Kepler)

**Comment by sthaha** [comment]
cc: @marceloamaral @rootfs @sunya-ch

**Reply by rootfs** [reply]
thank you @sthaha formulating the process
